### ISSUE TITLE
[Suggestion] KubeJS Support

### ISSUE THREAD

**Issue by Saereth** [Draylar/illusion#1]
This could benefit greatly from KubeJS support to setup the block builders with scripts instead of needing to write another mod for it. Additionally, add the predicate of player gamestage as supported by the gamestages mod (works with Kube as well) would be a hugely beneficial target for the illusion conditions of blocks. This would enable a sevtech style orestages use case.

**Comment by Draylar** [comment]
Hi Saereth, thanks for the issue post. Illusion was tailor-made for [Game Phases](https://github.com/Draylar/game-phases), which is a KubeJS-based stage manager that should cover the use cases you mentioned above. It will be available on 1.18.2/1.19.2 Forge within the next week. As an example, if you would like to hide Diamond Ore as Stone until the player unlocks an advancement, you could use the following KJS script:

```java
onEvent('gamephases.initialize', event => {
    // Replace stone with Diamond Ore
    event.phase('one');
        .block('minecraft:diamond_ore', 'minecraft:stone');
});
```
When you are ready to unlock this Illusion, you can either call `player.grantPhase("one")` via KJS, or use the advancement reward system if you already have something for that in place. Does this fit your needs, or are you looking for a more literal 1:1 API translation where you could do something like...

```java
illusion("minecraft:diamond_ore", "minecraft:stone", player => player.isSneaking())
```

I can look into implementing the second example, but I think it would be best put in a separate project (eg. Illusion KubeJS Support). Thanks again for the feedback and suggestions - much appreciated!

**Comment by Saereth** [comment]
Yeah that should work just fine. Cool and thanks for the response.
